src/MigrationTools.Clients.AzureDevops.ObjectModel/Enrichers/TfsWorkItemEmbededLinkEnricher.cs

```diff
@@ -0,0 +1,145 @@
+        }
+
+        [Obsolete]
+        public override int Enrich(WorkItemData sourceWorkItem, WorkItemData targetWorkItem)
+        {
+            string oldTfsurl = Engine.Source.Config.AsTeamProjectConfig().Collection.ToString();
+            string newTfsurl = Engine.Target.Config.AsTeamProjectConfig().Collection.ToString();
+
+            var logTypeName = nameof(TfsWorkItemEmbededLinkEnricher);
+            Log.LogInformation($"{logTypeName}: Fixing embedded mention links on target work item {targetWorkItem.Id} from {oldTfsurl} to {newTfsurl}");
+
+            foreach (Field field in targetWorkItem.ToWorkItem().Fields)
+            {
+                if (field.Value == null
+                    || string.IsNullOrWhiteSpace(field.Value.ToString())
+                    || (field.FieldDefinition.FieldType != FieldType.Html && field.FieldDefinition.FieldType != FieldType.History))
+                {
+                    continue;
+                }
+
+                try
+                {
+                    var anchorTagMatches = Regex.Matches((string)field.Value, RegexPatternLinkAnchorTag);
+                    foreach (Match anchorTagMatch in anchorTagMatches)
+                    {
+                        if (!anchorTagMatch.Success) continue;
+
+                        var href = anchorTagMatch.Groups["href"].Value;
+                        var version = anchorTagMatch.Groups["version"].Value;
+                        var value = anchorTagMatch.Groups["value"].Value;
+
+                        if (string.IsNullOrWhiteSpace(href) || string.IsNullOrWhiteSpace(version) || string.IsNullOrWhiteSpace(value))
+                            continue;
+
+                        var workItemLinkMatch = Regex.Match(href, RegexPatternWorkItemUrl);
+                        if (workItemLinkMatch.Success)
+                        {
+                            var workItemId = workItemLinkMatch.Groups["id"].Value;
+                            Log.LogDebug($"{logTypeName}: Source work item {workItemId} mention link traced on field {field.Name} on target work item {targetWorkItem.Id}.");
+                            var sourceLinkWi = Engine.Source.WorkItems.GetWorkItem(workItemId);
+                            if (sourceLinkWi != null)
+                            {
+                                var linkWI = Engine.Target.WorkItems.FindReflectedWorkItemByReflectedWorkItemId(sourceLinkWi);
+                                if (linkWI != null)
+                                {
+                                    var replaceValue = anchorTagMatch.Value.Replace(workItemId, linkWI.Id);
+                                    field.Value = field.Value.ToString().Replace(anchorTagMatch.Value, replaceValue);
+                                    Log.LogInformation($"{logTypeName}: Source work item {workItemId} mention link was successfully replaced with target work item {linkWI.Id} mention link on field {field.Name} on target work item {targetWorkItem.Id}.");
```

Comment:
All Logging should be in the format `"My message that contains {item} and {item2}", item, itesm2`! Do not use `$"My message that contains {item} and {item2}"` to pass text into the log strings as this disables [Serilog](https://serilog.net/)'s ability to pass that data as telemetry to Application Insights and for log highlighting.

Reply:
Updated..